From 3a1d6afccfd199f109998f57fb5f5b76c4999999 Mon Sep 17 00:00:00 2001
From: Build Fix Bot <noreply@example.com>
Date: Tue, 3 Mar 2026 13:10:00 +0300
Subject: [PATCH] fix: C++20-compatible FileWatch move ctor declaration

GCC 14/C++20 rejects constructor template-id syntax:
  FileWatch<T>(FileWatch<T>&&)
Use the C++20-compliant form:
  FileWatch(FileWatch&&)
---
 thirdparty/filewatch/FileWatch.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/thirdparty/filewatch/FileWatch.hpp b/thirdparty/filewatch/FileWatch.hpp
index aaae62f..11257e0 100644
--- a/thirdparty/filewatch/FileWatch.hpp
+++ b/thirdparty/filewatch/FileWatch.hpp
@@ -117,7 +117,7 @@ namespace filewatch {
         }
 
         // Const memeber varibles don't let me implent moves nicely, if moves are really wanted std::unique_ptr should be used and move that.
-        FileWatch<T>(FileWatch<T>&&) = delete;
+        FileWatch(FileWatch&&) = delete;
         FileWatch<T>& operator=(FileWatch<T>&&) & = delete;
 
     private:
-- 
2.43.0
